handlers/handler.go

```diff
@@ -26,6 +26,45 @@ func internalServerError(w http.ResponseWriter, err error) {
 	http.Error(w, errMsg, http.StatusInternalServerError)
 }
 
+func (env *Env) PostAuthHandler(w http.ResponseWriter, r *http.Request) {
+	reqUser := &models.User{}
+	if err := parseJSON(w, r.Body, reqUser); err != nil {
+		return
+	}
+	if reqUser.Email == "" || reqUser.Password == "" {
+		errMsg := "Request body is missing field(s)"
+		log.Println(errMsg)
+		http.Error(w, errMsg, http.StatusBadRequest)
+		return
+	}
+	user := &models.User{}
+	user, err := env.DB.CheckUser(reqUser)
+	if err != nil {
+		mySQLErr, ok := err.(*mysql.MySQLError)
+		if ok && mySQLErr.Number == 1065 {
+			errMsg := fmt.Sprintf("User with email %s was not found", reqUser.Email)
+			log.Println(errMsg)
+			http.Error(w, errMsg, http.StatusNotFound)
+		} else if err.Error() == "password incorrect" {
+			errMsg := fmt.Sprintf("Password incorrect")
+			log.Println(errMsg)
+			http.Error(w, errMsg, http.StatusUnauthorized)
```

Comment:
The capital `Password` isn't super necessary, so you can just use `err.Error()`.
```suggestion
			log.Println(err.Error())
			http.Error(w, err.Error(), http.StatusUnauthorized)
```